qutip/solver/solver_base.py

```diff
@@ -51,6 +51,7 @@ def __init__(self, rhs, *, options=None):
         self._integrator = self._get_integrator()
         self._state_metadata = {}
         self.stats = self._initialize_stats()
+        self.rhs._register_feedback({}, solver=self.name)
```

Comment:
Maybe `Solver._register_feedback` should be a method that solver instances can override to register their feedback? And it can be called whenever the RHS or equivalent are updated.

Reply:
No instance, but each solver subclass that has something to offer overwrite it.
It's called mostly in `set_state` since it must be called once per trajectories.

Reply:
I meant that if this call was `self._register_feedback()` rather than `self.rhs._register_feedback(...)` then each Solver class could update all the state it needed inside `_register_feedback` (e.g. add feedback to rhs and c_ops) and the feedback registration could all happen in one method.